src/constructor.io/client/modules/Catalog.cs

```diff
@@ -971,5 +971,260 @@ public async Task<SearchabilitiesResponse> PatchSearchabilities(PatchSearchabili
+        /// Retrieves a list of all Sort Options.
+        /// </summary>
+        /// <param name="sortOptionsRequest">Constructorio's <see cref="SortOptionsRequest"/> object model.</param>
+        /// <returns>Constructorio's <see cref="SortOptionList"/> object.</returns>
+        public async Task<SortOptionList> RetrieveSortOptions(SortOptionsRequest sortOptionsRequest)
+        {
+            string url = CreateSortOptionsUrl(sortOptionsRequest);
+            Dictionary<string, string> requestHeaders = new Dictionary<string, string>();
+            AddAuthHeaders(this.Options, requestHeaders);
+
+            string result;
+            try
+            {
+                result = await MakeHttpRequest(this.Options, HttpMethod.Get, url, requestHeaders);
+            }
+            catch (Exception e)
+            {
+                throw new ConstructorException(e);
+            }
+
+            if (result != null)
+            {
+                return JsonConvert.DeserializeObject<SortOptionList>(result);
+            }
+
+            throw new ConstructorException("RetrieveSortOptions response data is malformed");
+        }
+
+        /// <summary>
+        /// Retrieves a list of all Sort Options associated with a sort_by field.
+        /// </summary>
+        /// <param name="retrieveSortOptionBySortByRequest">Constructorio's <see cref="RetrieveSortOptionBySortByRequest"/> object model.</param>
+        /// <returns>Constructorio's <see cref="SortOptionList"/> object.</returns>
+        public async Task<SortOptionList> RetrieveSortOptionsBySortBy(RetrieveSortOptionBySortByRequest retrieveSortOptionBySortByRequest)
+        {
+            string url = CreateSortOptionsUrl(retrieveSortOptionBySortByRequest, retrieveSortOptionBySortByRequest.FilterBySortBy);
+            Dictionary<string, string> requestHeaders = new Dictionary<string, string>();
+            AddAuthHeaders(this.Options, requestHeaders);
+
+            string result;
+            try
+            {
+                result = await MakeHttpRequest(this.Options, HttpMethod.Get, url, requestHeaders);
+            }
+            catch (Exception e)
+            {
+                throw new ConstructorException(e);
+            }
+
+            if (result != null)
+            {
+                return JsonConvert.DeserializeObject<SortOptionList>(result);
+            }
+
+            throw new ConstructorException("RetrieveSortOptionsBySortBy response data is malformed");
+        }
+
+        /// <summary>
+        /// Replaces all existing Sort Options with a new list of Sort Options.
+        /// At most 50 Sort Options can be provided.
+        /// </summary>
+        /// <param name="sortOptionsListRequest">Constructorio's <see cref="SortOptionsListRequest"/> object model.</param>
+        /// <returns>Constructorio's <see cref="SortOptionList"/> object.</returns>
+        public async Task<SortOptionList> SetSortOptions(SortOptionsListRequest sortOptionsListRequest)
```

Comment:
Similar to my comment above, it seems a bit complex to need to keep track of all these request classes. Should we do something similar to how our facet functions work?

In this case `SetSortOptions` would accept two parameters, a list of sort options and a section (that defaults to 'Products')

Reply:
Following an in-depth discussion, we're running with the Request Object pattern. See [Slack Thread Here](https://constructor.slack.com/archives/C02RSEE9QE4/p1683741624059059).